packages/@lwc/babel-plugin-component/src/component.ts

```diff
@@ -1,34 +1,51 @@
 /*
- * Copyright (c) 2018, salesforce.com, inc.
+ * Copyright (c) 2023, salesforce.com, inc.
  * All rights reserved.
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-const { basename, extname } = require('path');
-
-const moduleImports = require('@babel/helper-module-imports');
-
-const {
+import { basename, extname } from 'path';
+import * as types from '@babel/types';
+import { addDefault, addNamed } from '@babel/helper-module-imports';
+import { NodePath } from '@babel/traverse';
+import { Visitor } from '@babel/core';
+import {
     LWC_PACKAGE_ALIAS,
     REGISTER_COMPONENT_ID,
     TEMPLATE_KEY,
     COMPONENT_NAME_KEY,
-} = require('./constants');
+} from './constants';
+import { BabelAPI, LwcBabelPluginPass } from './types';
 
-function getBaseName(classPath) {
+function getBaseName(classPath: string) {
     const ext = extname(classPath);
     return basename(classPath, ext);
 }
 
-function importDefaultTemplate(path, state) {
+function importDefaultTemplate(
+    path: NodePath<
+        | types.ClassDeclaration
+        | types.FunctionDeclaration
+        | types.TSDeclareFunction
+        | types.Expression
+    >,
```

Comment:
Nit: a type alias would be nice here. This is a bit gnarly to read.

Reply:
I was probably copy-pasting from the auto-generated TS types. Fixed.